packages/atomic/src/components/insight/atomic-insight-result-action/atomic-insight-result-action.tsx

```diff
@@ -1,11 +1,30 @@
-import {Component, Prop, State, h} from '@stencil/core';
-import Quickview from '../../../images/quickview.svg';
+import {Result} from '@coveo/headless';
+import {Component, Event, EventEmitter, Prop, State, h} from '@stencil/core';
+import AttachIcon from '../../../images/attach.svg';
+import CopyIcon from '../../../images/copy-dark.svg';
+import EmailIcon from '../../../images/email.svg';
+import QuickviewIcon from '../../../images/preview.svg';
+import FeedIcon from '../../../images/share-post.svg';
 import {
   InitializableComponent,
   InitializeBindings,
 } from '../../../utils/initialization-utils';
 import {IconButton} from '../../common/iconButton';
 import {Bindings} from '../../search/atomic-search-interface/atomic-search-interface';
+import {ResultContext} from '../../search/result-template-components/result-template-decorators';
+
+export interface InsightResultActionClickedEvent {
+  action: string;
+  result: Result;
+}
+
+enum Actions {
```

Comment:
I would suggest to refrain from using enums: They are a "weird"/un-necessary TypeScript concept that do not map super well to underlying generated JS code. VS a simple string.

```typescript
type Action = 'copyToClipboard' | 'attachToCase' | 'quickview' | 'postToFeed' | 'sendAsEmail';
```